### PR TITLE
Auth: Added support to filter for parent teams in GitHub connector's team membership filter

### LINKED ISSUE #85916: Auth: Add support to filter for parent teams in GitHub connector's team membership filter
Currently the GitHub connector's team membership filter only filters for teams that the currently authenticated user directly assigned to and doesn't consider the parent teams, so for example when someone specifies the parent team id in the `team_ids` then users won't be able to login if they are not directly assigned to the particular team on GitHub (they are assigned to one of the nested/sub teams).

Extend the GitHub connector team membership filter implementation to filter among the parent teams as well.

Relevant part:
https://github.com/grafana/grafana/blob/8461310e36395fad37cea4bb12c8fcb446b06806/pkg/login/social/connectors/github_oauth.go#L135-L154

More info about teams:
- [About teams](https://docs.github.com/en/organizations/organizing-members-into-teams/about-teams)
- [REST API endpoint docs](https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#list-teams-for-the-authenticated-user)

### PR DESCRIPTION
**Pull request by wasim-nihal**
**What is this feature?**

Added support to filter for parent teams in GitHub connector's team membership filter

**Why do we need this feature?**

To enable users to configure parent team id in the `team_ids` section of ini

**Who is this feature for?**

Users using `team_ids` for Github Oauth

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/85916

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by wasim-nihal** [comment]
@mgyongyosi could you please review and let me know your comments on this PR?

File: pkg/login/social/connectors/github_oauth.go [hunk 1]
```diff
@@ -45,6 +45,9 @@ type GithubTeam struct {
 	Organization struct {
 		Login string `json:"login"`
 	} `json:"organization"`
+	Parent struct {
```
Comment (mgyongyosi) [inline pkg/login/social/connectors/github_oauth.go:48]: The goal is to match the APIs contract and `Parent` can be null.
```suggestion
	Parent *struct {
```

File: pkg/login/social/connectors/github_oauth_test.go [hunk 2]
```diff
@@ -216,6 +250,29 @@ func TestSocialGitHub_UserInfo(t *testing.T) {
...
+				Name:   "monalisa octocat",
+				Email:  "octocat@github.com",
+				Login:  "octocat",
+				Role:   "Editor",
+				Groups: []string{"https://github.com/orgs/github/teams/justice-league", "@github/justice-league"},
+			},
+		},
+		{
+			// see: https://github.com/grafana/grafana/issues/85916
+			name:              "should not error out, when the team_id is of a parent team",
```
Comment (mgyongyosi) [inline pkg/login/social/connectors/github_oauth_test.go:268]:
```suggestion
			name:              "should check parent team id for team membership",
```

**Review by mgyongyosi — COMMENTED** [review]
Thanks @wasim-nihal! I left some feedback, could you please address those?

**Review by mgyongyosi — APPROVED** [review]
Thanks! It looks good! I left a small suggestion.

Please address the issue and rebase your branch on `master`!

**Comment by wasim-nihal** [comment]
> Thanks! It looks good! I left a small suggestion.
> 
> Please address the issue and rebase your branch on `master`!

@mgyongyosi , resolved the comment and rebased

**Comment by mgyongyosi** [comment]
Thank you for the contribution @wasim-nihal!

**Pull request merged**